bundles/org.openhab.binding.fronius/src/main/java/org/openhab/binding/fronius/internal/handler/FroniusBaseThingHandler.java

```diff
@@ -69,29 +71,16 @@ public void handleCommand(ChannelUID channelUID, Command command) {
 
     @Override
     public void initialize() {
-        if (getFroniusBridgeHandler() == null) {
-            logger.debug("Initializing {} Service is only supported within a bridge", serviceDescription);
-            updateStatus(ThingStatus.OFFLINE);
-            return;
-        }
         logger.debug("Initializing {} Service", serviceDescription);
-        getFroniusBridgeHandler().registerService(this);
-    }
-
-    private synchronized FroniusBridgeHandler getFroniusBridgeHandler() {
-        if (this.bridgeHandler == null) {
-            Bridge bridge = getBridge();
-            if (bridge == null) {
-                return null;
-            }
-            ThingHandler handler = bridge.getHandler();
-            if (handler instanceof FroniusBridgeHandler) {
-                this.bridgeHandler = (FroniusBridgeHandler) handler;
-            } else {
-                return null;
-            }
+        // this is important so FroniusBridgeHandler::childHandlerInitialized gets called
+        Bridge bridge = getBridge();
+        if (bridge == null) {
```

Comment:
Please also check that bridge.getHandler() is not null

Reply:
Added check for bridge.getHandler()